[inductor][ck] add kBatch_sweep to config.rocm

coconutruben opened the pull request:
Summary:
# Why

enable testing and users to specify a set of kBatches to try rather than relying on our hand written heuristic

# What

add rocm.kBatch_sweep as a list of kBatches to try out. These will generate a product of CK instances, one per kBatch for each existing op, though they are often filtered out if they are likely to fail at runtime

Test Plan: n/a

Reviewed By: chenyang78

Differential Revision: D70226055




cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @dllehr-amd @jataylo @hongxiayang @naromero77amd @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler commented on torch/_inductor/codegen/rocm/ck_universal_gemm_template.py:
```diff
@@ -887,8 +887,11 @@ def _get_kBatch(self, op):
+        if K // max(M, N) < config.rocm.split_k_threshold:
             return [1]
+        # if the user is telling us which kBatches to sweep, just use those
+        if config.rocm.kBatch_sweep is not None:
+            return config.rocm.kBatch_sweep
```
might makes sens to move this part before the above split_k_threshold check?

coconutruben replied:
how come? they're both parametrizable, so if the user wants this to happen on every case, they can set the threshold to be 0, but if we move it above, then we're forced to use the configured sweep for all cases regardless

ColinPeppler replied:
i saw it as, user might set the kbatch_sweep and tell inductor to just run with that (don't care about split_k_threshold) but I guess it makes sense to do the split_k_threshold check since it's protecting you from launching with a kbatch that doesn't make sense.

coconutruben committed:
[inductor][ck] add kBatch_sweep to config.rocm

Summary:
# Why

enable testing and users to specify a set of kBatches to try rather than relying on our hand written heuristic

# What

add rocm.kBatch_sweep as a list of kBatches to try out. These will generate a product of CK instances, one per kBatch for each existing op, though they are often filtered out if they are likely to fail at runtime

Test Plan: n/a

Reviewed By: chenyang78

Differential Revision: D70226055

ColinPeppler commented:
also may want to consider adding unittests for this
